backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/apiaudit/apiAuditPInstance.feature

```diff
@@ -0,0 +1,43 @@
+@from:cucumber
+Feature: API Audit
+
+  Background:
+    Given the existing user with login 'metasfresh' receives a random a API token for the existing role with name 'WebUI'
+    And all the API audit data is reset
+
+  @from:cucumber
+  Scenario: if a 'x-adpinstanceid' header is set when auditing v2 calls, it should end up in API_Request_Audit.AD_PInstance_ID
+    Given the following API_Audit_Config record is set
+      | Identifier | SeqNo | OPT.Method | OPT.PathPrefix | IsInvokerWaitsForResult |
+      | c_1        | 10    | POST       | api/v2/test    | Y                       |
+
+    And add I_AD_PInstance with id 12052022
+
+    And following http headers are set on context
+      | Name            | Value    |
+      | x-adpinstanceid | 12052022 |
+
+    When a 'POST' request with the below payload and headers from context is sent to the metasfresh REST-API 'api/v2/test?responseBody=%22test-endpoint%20was%20called%22&responseCode=200' and fulfills with '200' status code
+    And the actual response body is
+    """
+    {
+	"messageBody": "\"test-endpoint was called\""
+    }
+    """
+
+    And there are added records in API_Request_Audit
+      | OPT.AD_PInstance_ID | Method | Path                                                                          | AD_User.Name | Status      |
+      | 12052022            | POST   | /api/v2/test?responseBody=%22test-endpoint%20was%20called%22&responseCode=200 | metasfresh   | Verarbeitet |
+
+    And there are added records in API_Request_Audit_Log
+      | Logmessage                                | AD_Issue.Summary |
+      | Endpoint invoked; returning httpCode: 200 | null             |
+
+    And there are added records in API_Response_Audit
+      | HttpCode | Body                                           |
+      | 200      | {"messageBody":"\"test-endpoint was called\""} |
+
+    And all the API audit data is reset
+    And the following API_Audit_Config record is set
+      | Identifier | SeqNo | OPT.Method | OPT.PathPrefix | IsInvokerWaitsForResult |
+      | c_2        | 9980  | null       | null           | Y                       |
```

Comment:
now popping into my mind: how does the API react if there is no AD_PInstance with the given ID?
i think the API should then fail with a helpful error-message

Reply:
LE: such a fail with helpful error message should happen on any invalid value for any of our `X-` headers

Reply:
(created a follow-up issue for this)